src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs

```diff
@@ -2132,6 +2124,42 @@ private ReadOnlyCollection<TValue> GetValues()
             }
         }
 
+        /// <summary>Whether T's type can be written atomically (i.e., with no danger of torn reads).</summary>
```

Comment:
It feels like that this can produce pretty severe regression on Mono since it does not have the cache. We should either get some Mono perf numbers for this to convince us that it is ok; or you can just keep a simple bool cache under ifdef (like it existed before https://github.com/dotnet/runtime/commit/e8fa4a736d6bdcfbbe2c2dc1dd1303214bb97085)

```
#if MONO
// Cache the value on Mono
private static bool IsValueWriteAtomic() => s_isValueWriteAtomic;

private static bool s_isValueWriteAtomic = ComputeValueWriteAtomic();

private static bool ComputeValueWriteAtomic()
#else
// The whole method will be turned into a constant by the JIT
[MethodImpl(MethodImplOptions.AggressiveInlining)]
private static bool IsValueWriteAtomic()
#endif
```